src/main/webapp/app/complaints/complaints-for-students/complaints-student-view.component.ts

```diff
@@ -0,0 +1,160 @@
+            });
+        }
+        this.loadPotentialComplaint();
+        this.accountService.identity().then((user) => {
+            if (this.participation && this.participation.student && user?.id) {
+                this.isCurrentUserSubmissionAuthor = this.participation.student.id === user.id;
+            }
+        });
+
+        this.timeOfFeedbackRequestValid = this.isTimeOfFeedbackRequestValid();
+        this.timeOfComplaintValid = this.isTimeOfComplaintValid();
+        this.showComplaintsSection = this.isShowComplaintsSection();
+    }
+
+    /**
+     * Sets the complaint if a complaint and a valid result exists
+     */
+    loadPotentialComplaint(): void {
+        if (this.result && this.result.completionDate) {
+            this.complaintService
+                .findBySubmissionId(this.submission.id!)
+                .pipe(filter((res) => !!res.body))
+                .subscribe((res) => {
+                    this.complaint = res.body!;
+                });
+        }
+    }
+
+    /**
+     * This function is used to check whether the student is allowed to submit a complaint or not.
+     * For exams, submitting a complaint is allowed within the Student Review Period, see {@link isWithinExamReviewPeriod}.
+     *
+     * For normal course exercises, submitting a complaint is allowed within one week after the student received the
+     * result. If the result was submitted after the assessment due date or the assessment due date is not set, the completion date of the result is checked. If the result was
+     * submitted before the assessment due date, the assessment due date is checked, as the student can only see the result after the assessment due date.
+     */
+    private isTimeOfComplaintValid(): boolean {
+        if (this.result?.completionDate) {
+            if (this.isExamMode) {
+                return this.isWithinExamReviewPeriod();
+            }
+            return this.isCompletionDateWithinCourseReviewPeriod(this.result.completionDate);
+        }
+        return false;
+    }
+
+    /**
+     * Analogous to isTimeOfComplaintValid but exams cannot have more feedback requests.
+     */
+    private isTimeOfFeedbackRequestValid(): boolean {
+        if (!this.isExamMode && this.result?.completionDate) {
+            return this.isCompletionDateWithinCourseReviewPeriod(this.result.completionDate);
+        }
+        return false;
+    }
+
+    /**
+     * Determines whether or not to show the complaint section
+     */
+    private isShowComplaintsSection(): boolean {
+        if (this.isExamMode) {
+            return this.isTimeOfComplaintValid() || !!this.complaint;
+        } else {
+            return !!(this.course?.complaintsEnabled || this.course?.requestMoreFeedbackEnabled);
+        }
+    }
+
+    /**
+     * Checks if a complaint (either actual complaint or more feedback request) can be filed
+     */
+    private isCompletionDateWithinCourseReviewPeriod(completionDate: dayjs.Dayjs): boolean {
```

Comment:
Either the comment or the method name should indicate what is meant by completion. I think it is not 100% clear in this context.

Reply:
Would it be enough for you to specify that it's the completion date of the result?

Reply:
What is a result completion date? Has it to be within the course review period to file a complaint? I do not have much experience with this code, I am not used to the terminology.

Reply:
I renamed it to `canFileComplaintWithCompletionDate()` since it's not exclusively the completionDate.
I also added a comment.